controllers/limitador_controller.go

```diff
@@ -199,6 +202,30 @@ func (r *LimitadorReconciler) reconcileDeployment(ctx context.Context, limitador
 		return err
 	}
 
+	// TODO: This block of code to be deleted when the upgrade path is no longer needed.
+	{
```

Comment:
I would wrap this code in a function maybe in `pkg/upgrades/v0.4.0` package. Then method clearly belongs to a release and can be easily removed on  a future release, let's say v0.7.0, assuming nobody jumps from 0.4.0 to 0.7.0.

Reply:
Thank you for the comment. Added a new package as suggested: `pkg/upgrades/v0_6_0`